Docs/developer_guide/parameter_nodes/gui_connection.md

```diff
@@ -0,0 +1,192 @@
+# GUI Connection
+
+## Data binding
+
+The parameter node wrappers have the ability to do two-way data binding between particular GUI elements and parameters of certain types. This means that when the GUI is updated the parameter node wrapper will be automatically updated, and when the parameter node wrapper is updated the GUI will automatically be updated.
+
+There are two ways to declare which GUI widget elements connect to which parameters.
+
+## Connecting widgets from a .ui file
+
+Qt Designer has the ability to set Dynamic Properties. These can be used to inform the parameter node wrapper infrastructure which parameters to connect to which widgets. This is the preferred way of connection from the `.ui` file of a scripted module to its parameter node wrapper.
+
+Simply set a dynamic string property named "SlicerParameterName" to the parameter name it should bind with.
+
+![Qt Dynamic Property](gui_connection-QtDynamicProperty.png)
```

Comment:
```suggestion
![Qt Dynamic Property](https://github.com/Slicer/Slicer/releases/download/docs-resources/parameter_node_gui_connection_QtDynamicProperty.png)
```

Now that I uploaded the image to https://github.com/Slicer/Slicer/releases/tag/docs-resources, consider removing it from this pull-request.